lib/util/merge-sorted-arrays.ts

```diff
@@ -0,0 +1,39 @@
+import PriorityQueue from "js-priority-queue";
+
+interface Entry<T> {
+  array: T[];
+  index: number;
+}
+
+function mergeSortedArrays<
```

Comment:
Why can't we just merge then sort by block number after?

Reply:
this is more efficient on large datasets

Reply:
How much time does this save for a typical user? Seems a bit overkill to add a new library for this. I am a bit skeptical or whether this is actually more performant as it's still looping over both of the arrays

Reply:
![image](https://github.com/zeitgeistpm/ui/assets/551965/a22814e5-e043-459f-9ca0-8b19de499c15)

Reply:
![image](https://github.com/zeitgeistpm/ui/assets/551965/667c89a4-a163-4910-bd10-4a472cce1b7d)

Reply:
![image](https://github.com/zeitgeistpm/ui/assets/551965/d0fa4894-936f-48f2-bbda-a5b3c375bf72)

Reply:
It takes less than 1ms to sort an array of 10000 elements so I don't think this is something we need to optimse
<img width="316" alt="image" src="https://github.com/zeitgeistpm/ui/assets/4950844/7266bebb-2b9c-4916-8771-05db9e867e2d">